qiskit/synthesis/discrete_basis/gate_sequence.py

```diff
@@ -19,7 +19,7 @@
 import numpy as np
 
 from qiskit.circuit import Gate, QuantumCircuit, Qubit
-from qiskit.extensions import UnitaryGate
+from qiskit.circuit.library.generalized_gates.unitary import UnitaryGate
```

Comment:
Not too important, but imo this is is too specific an import - the documented location is `qiskit.circuit.library`, and hyperspecific import locations like this are part of the reason it's difficult to reorganise code in the future.  If it doesn't import without the hyperspecificity, it's a strong sign that the objects have been put in the wrong place, and there are true cyclic-import problems.

Reply:
This might again be a side effect of Pylint's cyclic import complaints...

Reply:
Yeah, I'm saying that in this case, this would be pylint flagging something that's actually a real problem with our logical organisation - `generalized_gates` should depend on `standard_gates`, not vice versa.  `UnitaryGate` is a core gate; it's used in the definition of `Gate` itself, so it probably would have been better for it to be given special treatment in the hierarchy, such as being in the top level of `qiskit.circuit.library` or the like.